Pod/Classes/LSClockState.m

```diff
@@ -82,18 +86,20 @@ - (void)addSampleWithOriginMicros:(SInt64)originMicros
         latestOffsetMicros = ((receiveMicros - originMicros) + (transmitMicros - destinationMicros)) / 2;
     }
 
-    // Discard the oldest sample and push the new one.
-    [self.samples removeObjectAtIndex:0];
-    [self.samples
-        addObject:[[LSSyncSample alloc] initWithDelayMicros:latestDelayMicros offsetMicros:latestOffsetMicros]];
+    dispatch_async(self.samplesQueue, ^{
```

Comment:
Ah, that makes perfect sense. I should have read the init code more carefully.

This is fine with me. Let's just add:

```
// Note that self.samplesQueue is DISPATCH_QUEUE_SERIAL, so there is no risk that two threads concurrently access `self.samples`.
```